data/src/main/scala/ch.epfl.scala.index.data/central/CentralMissing.scala

```diff
@@ -169,135 +145,97 @@ class CentralMissing(paths: DataPaths)(implicit val system: ActorSystem) {
+    )
+    DownloadRequest(groupId, artifactId, version, created)
   }
 
-  def savePomsAndMeta(
-      in: Either[String, (PomContent, DownloadRequest)]
-  ): Unit = {
-    in match {
-      case Left(failure) =>
-        log.error(failure)
-
-      case Right((pom, request)) => {
-        val sha1 = Sha1(pom.content)
-        val repository = LocalPomRepository.MavenCentral
+  def savePomsAndMeta(pom: PomContent): Unit = {
+    val sha1 = Sha1(pom.content)
+    val repository = LocalPomRepository.MavenCentral
 
-        val meta = Meta(sha1, request.path, request.created)
+    val meta = Meta(sha1, pom.artifact.path, pom.artifact.created)
 
-        // write meta
-        Meta.append(
-          paths,
-          meta,
-          repository
-        )
+    // write meta
+    Meta.append(
+      paths,
+      meta,
+      repository
+    )
 
-        // write pom
-        val pomPath = paths.poms(repository).resolve(s"$sha1.pom")
-        Files.write(pomPath, pom.content.getBytes(StandardCharsets.UTF_8))
-      }
-    }
+    // write pom
+    val pomPath = paths.poms(repository).resolve(s"$sha1.pom")
+    Files.write(pomPath, pom.content.getBytes(StandardCharsets.UTF_8))
   }
 
   // data/run central /home/gui/scaladex/scaladex-contrib /home/gui/scaladex/scaladex-index /home/gui/scaladex/scaladex-credentials
   def run(): Unit = {
     val artifactMetaExtractor = new ArtifactMetaExtractor(paths)
-    val releases: Set[(String, String)] =
+    val allGroups: Set[String] =
       PomsReader(LocalPomRepository.MavenCentral, paths)
         .load()
-        .collect { case Success((pom, _, _)) =>
-          artifactMetaExtractor(pom).flatMap(meta =>
-            if (meta.scalaTarget.isDefined && !meta.isNonStandard)
-              Some((pom.groupId, meta.artifactName))
-            else None
-          )
+        .flatMap {
+          case Success((pom, _, _)) =>
+            artifactMetaExtractor(pom)
+              .filter { meta =>
+                meta.scalaTarget.isDefined && !meta.isNonStandard
+              }
+              .map(_ => pom.groupId)
+          case _ => None
         }
-        .flatten
         .toSet
 
-    val allTargets = List(
-      ScalaJvm(ScalaVersion.`2.13`),
-      ScalaJvm(ScalaVersion.`2.12`),
-      ScalaJvm(ScalaVersion.`2.11`),
-      ScalaJvm(ScalaVersion.`2.10`),
-      SbtPlugin(ScalaVersion.`2.10`, Sbt.`0.13`),
-      SbtPlugin(ScalaVersion.`2.12`, Sbt.`1.0`),
-      ScalaJs(ScalaVersion.`2.13`, Js.`0.6`),
-      ScalaJs(ScalaVersion.`2.12`, Js.`0.6`),
-      ScalaJs(ScalaVersion.`2.11`, Js.`0.6`),
-      ScalaJs(ScalaVersion.`2.10`, Js.`0.6`),
-      ScalaNative(ScalaVersion.`2.11`, Native.`0.3`)
-    )
-
-    val releasesDownloads =
-      releases.flatMap { case (groupId, artifact) =>
-        allTargets.map(target =>
-          ArtifactRequest(groupId, artifact + target.encode)
-        )
-      }.toList
+    val releasesDownloads = allGroups.toList.map(SearchRequest(_))
```

Comment:
IIUC, we only perform requests to find Scala 3 artifacts belonging to an organization for which there are already releases indexed in scaladex?

I guess this is also what you mean by this comment in the description:

> One limitation is that all artifacts published with a new organization name are missed.

Reply:
Exact